src/utils/SecurityUtils.js

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+const _findById = (array, idToFind) => {
+  return array.find((element) => element.id === idToFind);
+};
+
+const areAccessControlListsIdentical = (aclA, aclB) => {
+  if (Object.keys(aclA).length !== Object.keys(aclB).length) return false;
+  for (const userAccessA of aclA) {
+    const userAccessB = _findById(aclB, userAccessA.id);
+    if (userAccessB === undefined || userAccessA.role !== userAccessB.role) return false;
+  }
+  return true;
+};
+
+const compareAccessControlLists = (currentACL, newACL) => {
+  const entriesToAdd = [];
+  const entriesToRemove = [];
+  for (const newUserAccess of newACL) {
```

Comment:
for be more readable, maybe we can create a function like getMissingUsersAccess and getUsersRoleChanged.
the result will be :
const newUsersAccess = getMissingUsersAccess(currentAcl, newAcl);
const deletedUsersAccess = getMissingUsersAccess(newAcl, currentAcl);
const userAcccessRoleChanged = getUsersAccessRoleChanged(currentAcl, newAcl);

const entriesToAdd = newUsersAccess.concat(userAcccessRoleChanged);
const entriesToRemove = deletedUsersAccess.concat(userAccessRoleChanged);

Reply:
Lisibility has been improved, reduced returned objects to only two :slightly_smiling_face: